src/Common/tests/System/PlatformDetection.cs

```diff
@@ -34,6 +34,58 @@ public static class PlatformDetection
 
         public static int WindowsVersion { get; } = GetWindowsVersion();
 
+        private static int s_isWinRT = -1;
```

Comment:
thanks @stephentoub.  The other idea could be defining it as Nullable and initialize it in the definition:

```C#
static bool? s_isWinRT = InitWinRT();
```

This will guard against the race condition